frontend/src/metabase/dashboard/components/AddSeriesModal/AddSeriesModal.tsx

```diff
@@ -1,39 +1,59 @@
+import { getIn } from "icepick";
 import { Component } from "react";
-import PropTypes from "prop-types";
 import { t } from "ttag";
-import { getIn } from "icepick";
-import { connect } from "react-redux";
-
-import Visualization from "metabase/visualizations/components/Visualization";
 
+import {
+  Card,
+  CardId,
+  DashCardId,
+  DashboardOrderedCard,
+  DatasetData,
+} from "metabase-types/api";
 import * as MetabaseAnalytics from "metabase/lib/analytics";
 import { color } from "metabase/lib/colors";
-import { loadMetadataForQueries } from "metabase/redux/metadata";
+import Visualization from "metabase/visualizations/components/Visualization";
 
 import { QuestionList } from "./QuestionList";
 
-class AddSeriesModal extends Component {
-  constructor(props, context) {
+interface Props {
+  dashcard: DashboardOrderedCard;
+  dashcardData: Record<DashCardId, Record<CardId, DatasetData>>;
+  fetchCardData: (
+    card: Card,
+    dashcard: DashboardOrderedCard,
+    options: {
+      clearCache?: boolean;
+      ignoreCache?: boolean;
+      reload?: boolean;
+    },
+  ) => Promise<unknown>;
+  setDashCardAttributes: (options: {
+    id: DashCardId;
+    attributes: Partial<DashboardOrderedCard>;
+  }) => void;
+  onClose: () => void;
+}
+
+interface State {
+  error: unknown;
+  isLoading: boolean;
+  series: NonNullable<DashboardOrderedCard["series"]>;
+}
+
+class AddSeriesModal extends Component<Props, State> {
+  constructor(props: Props, context: unknown) {
     super(props, context);
 
     this.state = {
       error: null,
       series: props.dashcard.series || [],
-      isLoadingMetadata: false,
+      isLoading: false,
```

Comment:
I refactored the old `this.state.state` to `isLoading`, since now it only has 2 states (previously there was `"incompatible"` state as well).